op-node/rollup/derive/channel_frame.go

```diff
@@ -91,21 +77,20 @@ func (f *Frame) UnmarshalBinary(r ByteReader) error {
 	if err != nil {
 		return fmt.Errorf("error reading ID: %w", err)
 	}
-	f.ID.Time, err = binary.ReadUvarint(r)
-	if err != nil {
-		return fmt.Errorf("error reading ID.Time: %w", err)
+	if err := binary.Read(r, binary.BigEndian, &f.ID.Time); err != nil {
```

Comment:
This is cool, didn't know about `binary.Read`. I've always used `binary.BigEndian.ReadUint64`, etc.

Reply:
I can't actually find a `binary.BigEndian.ReadUint64` which works on readers, a lot of the binary encoding API works on buffers